src/test/fuzz/txrequest.cpp

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <primitives/transaction.h>
+#include <random.h>
+#include <txrequest.h>
+#include <test/fuzz/fuzz.h>
+#include <crypto/common.h>
+#include <crypto/siphash.h>
+
+#include <bitset>
+#include <cstdint>
+#include <queue>
+#include <vector>
+
+namespace {
+
+constexpr int MAX_TXHASHES = 16;
+constexpr int MAX_PEERS = 16;
+
+//! Randomly generated GenTxids used in this test (length is MAX_TX).
+uint256 TXHASHES[MAX_TXHASHES];
+
+/** Precomputed random durations (positive and negative, each ~exponentially distributed). */
+std::chrono::microseconds DELAYS[256];
+
+struct Initializer
+{
+    Initializer()
+    {
+        // Use deterministic RNG to fill in txids and delays.
+        // Non-determinism hurts fuzzing.
+        FastRandomContext rng(true);
+        for (int txhash = 0; txhash < MAX_TXHASHES; txhash += 1) {
+            do {
+                TXHASHES[txhash] = rng.rand256();
+            } while (*(TXHASHES[txhash].begin() + 31) != txhash || *(TXHASHES[txhash].begin()) != txhash);
+        }
+        for (int i = 0; i < 16; ++i) {
+            DELAYS[i] = std::chrono::microseconds{i};
+        }
+        for (int i = 16; i < 128; ++i) {
+            DELAYS[i] = DELAYS[i - 1] + std::chrono::microseconds{1 + rng.randbits(((i - 10) * 2) / 9)};
+        }
+        for (int i = 128; i < 256; ++i) {
+            DELAYS[i] = -DELAYS[255 - i];
+        }
```

Comment:
8d0dd46b4fcdf4c664747e638eda046f8d51079c
Could the only `int i{0};` before all of the `for` loops be more readable and maintainable?

Reply:
Done.